music/src/gansynth/model.ts

```diff
@@ -0,0 +1,292 @@
+
+      this.outputShape = this.nn.outputShape as number[];
+      this.setWeights(vars);
+      console.log(`Output Shape:${this.outputShape}`);
+    });
+  }
+
+  private setWeights(vars: tf.NamedTensorMap) {
+    function getVar(name: string) {
+      const v = vars[name];
+      if (v === undefined) {
+        throw Error(`Variable not found: ${name}`);
+      }
+      return v;
+    }
+
+    // tslint:disable-next-line
+    const prefix = 'Generator/progressive_gan_generator/progressive_gan_block_';
+    const weights = [
+      getVar(`${prefix}1/conv0/conv2d/kernel/ExponentialMovingAverage`),
+      getVar(`${prefix}1/conv0/bias/ExponentialMovingAverage`),
+      getVar(`${prefix}1/conv1/conv2d/kernel/ExponentialMovingAverage`),
+      getVar(`${prefix}1/conv1/bias/ExponentialMovingAverage`),
+      getVar(`${prefix}2/conv0/conv2d/kernel/ExponentialMovingAverage`),
+      getVar(`${prefix}2/conv0/bias/ExponentialMovingAverage`),
+      getVar(`${prefix}2/conv1/conv2d/kernel/ExponentialMovingAverage`),
+      getVar(`${prefix}2/conv1/bias/ExponentialMovingAverage`),
+      getVar(`${prefix}3/conv0/conv2d/kernel/ExponentialMovingAverage`),
+      getVar(`${prefix}3/conv0/bias/ExponentialMovingAverage`),
+      getVar(`${prefix}3/conv1/conv2d/kernel/ExponentialMovingAverage`),
+      getVar(`${prefix}3/conv1/bias/ExponentialMovingAverage`),
+      getVar(`${prefix}4/conv0/conv2d/kernel/ExponentialMovingAverage`),
+      getVar(`${prefix}4/conv0/bias/ExponentialMovingAverage`),
+      getVar(`${prefix}4/conv1/conv2d/kernel/ExponentialMovingAverage`),
+      getVar(`${prefix}4/conv1/bias/ExponentialMovingAverage`),
+      getVar(`${prefix}5/conv0/conv2d/kernel/ExponentialMovingAverage`),
+      getVar(`${prefix}5/conv0/bias/ExponentialMovingAverage`),
+      getVar(`${prefix}5/conv1/conv2d/kernel/ExponentialMovingAverage`),
+      getVar(`${prefix}5/conv1/bias/ExponentialMovingAverage`),
+      getVar(`${prefix}6/conv0/conv2d/kernel/ExponentialMovingAverage`),
+      getVar(`${prefix}6/conv0/bias/ExponentialMovingAverage`),
+      getVar(`${prefix}6/conv1/conv2d/kernel/ExponentialMovingAverage`),
+      getVar(`${prefix}6/conv1/bias/ExponentialMovingAverage`),
+      getVar(`${prefix}7/conv0/conv2d/kernel/ExponentialMovingAverage`),
+      getVar(`${prefix}7/conv0/bias/ExponentialMovingAverage`),
+      getVar(`${prefix}7/conv1/conv2d/kernel/ExponentialMovingAverage`),
+      getVar(`${prefix}7/conv1/bias/ExponentialMovingAverage`),
+      getVar(`${prefix}7/to_rgb/conv2d/kernel/ExponentialMovingAverage`),
+      getVar(`${prefix}7/to_rgb/bias/ExponentialMovingAverage`),
+    ];
+    this.nn.setWeights(weights);
+  }
+
+  /**
+   * Predicts MelLogMagnitudes and IFreq from latent (z) and pitch conditioning.
+   *
+   * @param inputs A 4-D Tensor of latent (z) concatenated with one-hot pitch
+   * conditioning. Size [batch, 1, 1, N_LATENTS].
+   * @param batchSize Size of input batch.
+   */
+  predict(inputs: tf.Tensor4D, batchSize: number) {
```

Comment:
As a user I need to know the value of `N_LATENTS` as well as the pitch range of the one-hot encoding in order to form the inputs tensor. I'd prefer that to be encapsulated by the model.

`randomSample` is the right level of abstraction IMO, but for doing interpolations I need to use `predict` and these low-level details come into play.